cmd/eksctl-anywhere/cmd/renewcertificates_test.go

```diff
@@ -0,0 +1,381 @@
+		})
+	}
+}
+
+func validateRenewCertificatesOptions(_ *cobra.Command, rc *renewCertificatesOptions) error {
+	if rc.configFile == "" {
+		return fmt.Errorf("must specify --config")
+	}
+	return nil
+}
+
+// TestValidateComponent tests the validateComponent function.
+func TestValidateComponent(t *testing.T) {
+	tests := []struct {
+		name        string
+		component   string
+		expectError bool
+		errorMsg    string
+	}{
+		{
+			name:        "valid etcd component",
+			component:   "etcd",
+			expectError: false,
+		},
+		{
+			name:        "valid control-plane component",
+			component:   "control-plane",
+			expectError: false,
+		},
+		{
+			name:        "empty component",
+			component:   "",
+			expectError: false,
+		},
+		{
+			name:        "invalid component",
+			component:   "invalid",
+			expectError: true,
+			errorMsg:    "invalid component",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := validateComponent(tt.component)
+			if tt.expectError && err == nil {
+				t.Error("expected error but got none")
+			}
+			if !tt.expectError && err != nil {
+				t.Errorf("unexpected error: %v", err)
+			}
+			if tt.expectError && err != nil && !strings.Contains(err.Error(), tt.errorMsg) {
+				t.Errorf("expected error message to contain %q, got %q", tt.errorMsg, err.Error())
+			}
+		})
+	}
+}
+
+// createConfigFileFromYAML creates a temporary config file with the given YAML content.
+func createConfigFileFromYAML(t *testing.T, yamlContent string) (string, func()) {
+	tmpfile, err := os.CreateTemp("", "config-*.yaml")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := tmpfile.Write([]byte(yamlContent)); err != nil {
+		t.Fatal(err)
+	}
+	if err := tmpfile.Close(); err != nil {
+		t.Fatal(err)
+	}
+
+	return tmpfile.Name(), func() { os.Remove(tmpfile.Name()) }
+}
+
+// setupSSHKeyFile creates a temporary SSH key file for testing.
+func setupSSHKeyFile(t *testing.T) func() {
+	keyFile := "/tmp/test-key"
+	if err := os.WriteFile(keyFile, []byte("test-key"), 0o600); err != nil {
+		t.Fatal(err)
+	}
+	return func() { os.Remove(keyFile) }
+}
+
+// TestRenewCertificates tests the renewCertificates method.
+func TestRenewCertificates(t *testing.T) {
+	// Setup SSH key file
+	cleanup := setupSSHKeyFile(t)
+	defer cleanup()
+
+	// Define test cases
+	tests := []struct {
+		name        string
+		configFile  string
+		component   string
+		configYaml  string
+		expectError bool
+		errorMsg    string
+	}{
+		{
+			name:        "valid config with valid component",
+			component:   "etcd",
+			expectError: false,
+			configYaml: `
+clusterName: test-cluster
+controlPlane:
+  nodes:
+  - 192.168.1.10
+  os: ubuntu
+  sshKey: /tmp/test-key
+  sshUser: ec2-user
+etcd:
+  nodes:
+  - 192.168.1.20
+  os: ubuntu
+  sshKey: /tmp/test-key
+  sshUser: ec2-user
+`,
+		},
+		{
+			name:        "valid config with empty component",
+			component:   "",
+			expectError: false,
+			configYaml: `
```

Comment:
seems we define configYaml at multiple places in this test file. Is it possible to create a variable in this file which can be used in all the tests? like validConfigYaml, invalidConfigYaml?